src/Suggestor/EditorSuggestorPopup.ts

```diff
@@ -73,6 +89,17 @@ export class EditorSuggestor extends EditorSuggest<SuggestInfoWithContext> {
             (editor as any)?.cm?.contentDOM?.dispatchEvent(eventClone);
             return;
         }
+
+        if (value.taskItDependsOn != null) {
+            const newTask = ensureTaskHasId(
+                value.taskItDependsOn,
+                this.plugin.getTasks().map((task) => task.id),
+            );
+            value.appendText += ` ${newTask.id}`;
+
+            replaceTaskWithTasks({ originalTask: value.taskItDependsOn, newTasks: newTask });
+        }
+
```

Comment:
I spent an hour or two experimenting with this tonight, and I think it is going to be really very useful.

I tested it with both Emoji and Dataview formats.

There is one behavioural problem I found:

- If adding a dependency on a task in the same file (either above or below), Obsidian generates a warning about having to merge edits.
    - Observed in Source mode and Live Preview
    - This does not happen when the same edit is done via the modal

<img width="280" alt="image" src="https://github.com/obsidian-tasks-group/obsidian-tasks/assets/4840096/d34a7236-f5f1-41a6-8b88-167167c0478e">

The file is updated correctly, but the warning is irritating, and it is likely we will get bug reports about it.

The problem may be in this area.

Reply:
The issue arises in my understanding when the "Depends on" suggestion generates a new ID for the suggested task, which happens when the task suggested is in the same file and has no ID. This triggers the updating of the task and corresponding file via the "replaceTaskWithTasks" function. Obsidian detects the file change in its directory and attempts to merge it with the file displayed in the editor, resulting in the warning. The Pop Up, it behaves similarly to when the file is not in focus, such as when the modal is in focus instead of the editor, no Warning Apears. This behavior may be related to Obsidian's caching.

The likely culprit is the "replaceTaskWithTasks" function, which includes a call to vault.modify.
First solution option could be to replace the "replaceTaskWithTasks" call with a call to the [Editor API](https://docs.obsidian.md/Reference/TypeScript+API/Editor/replaceRange) within the same file to avoid the reloading merge. (Preassumption, not tested yet)
 
A second solution Option could be to exclude tasks without an ID from being displayed in the suggestor to prevent the issue altogether.

Will look into the First one when i have Time for it.

Reply:
For the Time Being, I found a Second Problem:
![image](https://github.com/obsidian-tasks-group/obsidian-tasks/assets/16257607/4d61ac47-f4cc-4cb6-b6a0-898089be00ad)

An Empty Task Property Symbol (Or Wrongly Filled Out), behind the ID Symbol, causes Problems with the ID Detection.
The Plugin doesn't Recognize the ID, and just appends a new One.

Should I Investigate this Behavior or is it User Error, for the Time Being?

Reply:
> An Empty Task Property Symbol (Or Wrongly Filled Out), behind the ID Symbol, causes Problems with the ID Detection.
> 
> The Plugin doesn't Recognize the ID, and just appends a new One.
> 
> Should I Investigate this Behavior or is it User Error, for the Time Being?

No, that's already documented behaviour - see [What do I need to know about the order of items in a task?](https://publish.obsidian.md/tasks/Editing/Auto-Suggest#What+do+I+need+to+know+about+the+order+of+items+in+a+task%3F).

Reply:
> The issue arises in my understanding when the "Depends on" suggestion generates a new ID for the suggested task, which happens when the task suggested is in the same file and has no ID. This triggers the updating of the task and corresponding file via the "replaceTaskWithTasks" function. Obsidian detects the file change in its directory and attempts to merge it with the file displayed in the editor, resulting in the warning. The Pop Up, it behaves similarly to when the file is not in focus, such as when the modal is in focus instead of the editor, no Warning Apears. This behavior may be related to Obsidian's caching.
> 
> The likely culprit is the "replaceTaskWithTasks" function, which includes a call to vault.modify. First solution option could be to replace the "replaceTaskWithTasks" call with a call to the [Editor API](https://docs.obsidian.md/Reference/TypeScript+API/Editor/replaceRange) within the same file to avoid the reloading merge. (Preassumption, not tested yet)

That is a very good summary, and matches my understanding of the behaviour.

The thing is, it also matches my understanding of the behaviour when the same dependency edit is made via the Edit Task modal - except that no warning is given there.

We _did_ see that behaviour during the development of editing dependencies via the modal, and it was fixed by adding some missing `await` lines. No idea if that is a red herring or not...

> A second solution Option could be to exclude tasks without an ID from being displayed in the suggestor to prevent the issue altogether.

That would greatly reduce the usefulness of this PR, I feel.


> Will look into the First one when i have Time for it.

Thank you.

Reply:
> We did see that behaviour during the development of editing dependencies via the modal, and it was fixed by adding some missing await lines. No idea if that is a red herring or not...

Tried it couldn't get it too work.
But I'm not very experienced with async/await.

Solved it via a Hotfix as a Replace in the Editor instead of the File Context.

The Behavior of the "replaceTaskWithTasks" function and Error Handling is very weird, and should Probably be added to the List, that should sooner or later be refactored.

Reply:
Thanks for fixing the 'file has been modified externally'-type message...

> The Behavior of the "replaceTaskWithTasks" function and Error Handling is very weird, and should Probably be added to the List, that should sooner or later be refactored.

Yes, absolutely.... Before doing this, we need to get it covered by automated tests, which is not proving to be easy.

Resolving.